### PR TITLE
feat(replays): Create a Replay specific SearchBox for the index page

### LINKED ISSUE #37968: List page search typeahead should be tailored to fields that are available inside replays
Right now when you open the search box on the replay list page there are all kinds of suggestions. Most of these fields are not available on replays. 

Image

### PR DESCRIPTION
**Pull request by ryan953**
Fixes #37968

### CODE REVIEW

File: static/app/utils/fields/index.ts [hunk 1]
```diff
@@ -1041,6 +1047,128 @@ export const DISCOVER_FIELDS = [
...
+  COUNT_SEGMENTS = 'countSegments',
+  // COUNT_URLS = 'countUrls',
+  DEVICE_MODEL = 'device.model',
+  DURATION = 'duration',
+  // ERROR_IDS = 'errorIds',
+  // LONGEST_TRANSACTION = 'longestTransaction',
+  OS_NAME = 'os.name',
+  OS_VERSION = 'os.version',
+  RELEASES = 'releases',
+  // TRACE_IDS = 'traceIds',
```
Comment (ryan953) [inline static/app/utils/fields/index.ts:1063]: leaving a few things in here which could be good to support

Maybe we can throw them onto the list here: https://github.com/getsentry/sentry/pull/40691/files
cc @JoshFerge

File: static/app/utils/replays/replayDataUtils.tsx [hunk 2]
```diff
@@ -41,16 +41,18 @@ export function mapResponseToReplayRecord(apiResponse: any): ReplayRecord {
...
-    ...(apiResponse.os?.name ? {'os.name': [apiResponse.os.name]} : {}),
-    ...(apiResponse.os?.version ? {'os.version': [apiResponse.os.version]} : {}),
     ...(apiResponse.browser?.name ? {'browser.name': [apiResponse.browser.name]} : {}),
     ...(apiResponse.browser?.version
       ? {'browser.version': [apiResponse.browser.version]}
       : {}),
-    ...(apiResponse.device?.name ? {'device.name': [apiResponse.device.name]} : {}),
-    ...(apiResponse.device?.family ? {'device.family': [apiResponse.device.family]} : {}),
     ...(apiResponse.device?.brand ? {'device.brand': [apiResponse.device.brand]} : {}),
+    ...(apiResponse.device?.family ? {'device.family': [apiResponse.device.family]} : {}),
```
Comment (billyvg) [inline static/app/utils/replays/replayDataUtils.tsx:49]: Hmm, I wonder if our backend should be flattening these keys for us. Our other APIs I think do e.g. the response will have a key of `device.family`.

File: static/app/views/replays/filters.tsx [hunk 3]
```diff
@@ -1,34 +1,47 @@
...
+import ReplaySearchBar from 'sentry/views/replays/replaySearchBar';
 
-type Props = {
-  handleSearchQuery: (query: string) => void;
-  organization: Organization;
-  query: string;
-};
+type Props = {};
+
+function ReplaysFilters({}: Props) {
```
Comment (billyvg) [inline static/app/views/replays/filters.tsx:17]: Is this needed?

```suggestion
function ReplaysFilters() {
```

**Review by billyvg — APPROVED** [review]

File: static/app/utils/replays/replayDataUtils.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/utils/replays/replayDataUtils.tsx:49]: This code is flattening the key, but also sticking the value into the `tags` object. 

So we start with something like:
```
const apiResponse = {
  id: "56832378a54f49dba046f909f80cb17f",
  startedAt: "2022-10-31T11:44:12+00:00"
  device: {
    family: "Mac"
  },
  tags: {}
};
```

and end with:
```
return {
  id: "56832378a54f49dba046f909f80cb17f",
  startedAt: "2022-10-31T11:44:12+00:00"
  device: {
    family: "Mac"
  },
  tags: {
    "device.family": "Mac"
  }
};
```

that makes it easier to populate the tags table later on

**Comment by ryan953** [comment]
I separated out some code that implements the `getTagValues` callback. This is because the backend right now has an error related to the `DataSets.REPLAYS` key. 
We'll bring it all back with https://github.com/getsentry/sentry/pull/40768/files

**Pull request merged**